src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  isInvalidTriangle(): boolean {
+    return this.a === 0
+      || this.b === 0
+      || this.c === 0
+      || this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b;
+  }
```

Comment:
With destructurization there will be no need to use 'this' so many times. You can try it.
**const { a, b, c } = this;**

Reply:
Thank you!